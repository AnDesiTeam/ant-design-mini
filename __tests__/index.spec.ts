/** @format */
import { Parser } from "../scripts/docs/parser/parse";

describe("add function", () => {
  it("parse comment", () => {
    const parser = new Parser("<!-- d -->");
    expect(JSON.stringify(parser.fragment)).toMatchSnapshot();
  });
  it("parse simple html", () => {
    const parser = new Parser(`<div d a="b" c="d" >fdsa</div>`);
    expect(JSON.stringify(parser.fragment)).toMatchSnapshot();
  });
  it("parse simple html", () => {
    const parser = new Parser("<div>{abc} {def}</div>");
    expect(JSON.stringify(parser.fragment)).toMatchSnapshot();
  });
  it("parse simple html", () => {
    const parser = new Parser(`<div name="test1 {test2}"></div>`);
    expect(JSON.stringify(parser.fragment)).toMatchSnapshot();
  });
  it("parse simple html", () => {
    const parser = new Parser(`<div name="test1 { a && b ? 'c' : '' }"></div>`);
    expect(JSON.stringify(parser.fragment)).toMatchSnapshot();
  });
});
