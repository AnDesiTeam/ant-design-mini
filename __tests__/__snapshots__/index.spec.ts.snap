// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add function parse comment 1`] = `"{\\"tagName\\":\\"html\\",\\"start\\":0,\\"children\\":[{\\"start\\":0,\\"end\\":10,\\"tagName\\":\\"Comment\\",\\"content\\":\\"<!-- d -->\\",\\"children\\":[],\\"attribute\\":[]}],\\"end\\":10,\\"attribute\\":[],\\"content\\":\\"<!-- d -->\\"}"`;

exports[`add function parse simple html 1`] = `"{\\"tagName\\":\\"html\\",\\"start\\":0,\\"children\\":[{\\"start\\":0,\\"end\\":30,\\"tagName\\":\\"div\\",\\"content\\":\\"<div d a=\\\\\\"b\\\\\\" c=\\\\\\"d\\\\\\" >fdsa</div>\\",\\"children\\":[{\\"start\\":20,\\"end\\":24,\\"type\\":\\"Text\\",\\"content\\":\\"fdsa\\"}],\\"attribute\\":[{\\"start\\":5,\\"attrName\\":\\"d\\",\\"attrValue\\":true,\\"end\\":6},{\\"start\\":7,\\"attrName\\":\\"a\\",\\"attrValue\\":[{\\"start\\":0,\\"end\\":1,\\"type\\":\\"Text\\",\\"content\\":\\"b\\"}],\\"end\\":12},{\\"start\\":13,\\"attrName\\":\\"c\\",\\"attrValue\\":[{\\"start\\":0,\\"end\\":1,\\"type\\":\\"Text\\",\\"content\\":\\"d\\"}],\\"end\\":18}]}],\\"end\\":30,\\"attribute\\":[],\\"content\\":\\"<div d a=\\\\\\"b\\\\\\" c=\\\\\\"d\\\\\\" >fdsa</div>\\"}"`;

exports[`add function parse simple html 2`] = `"{\\"tagName\\":\\"html\\",\\"start\\":0,\\"children\\":[{\\"start\\":0,\\"end\\":22,\\"tagName\\":\\"div\\",\\"content\\":\\"<div>{abc} {def}</div>\\",\\"children\\":[{\\"tagName\\":\\"Mustache\\",\\"start\\":5,\\"end\\":10,\\"children\\":[],\\"attribute\\":[],\\"content\\":\\"{abc}\\",\\"expression\\":{\\"type\\":\\"Identifier\\",\\"start\\":0,\\"end\\":3,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":0},\\"end\\":{\\"line\\":1,\\"column\\":3}},\\"name\\":\\"abc\\"}},{\\"start\\":10,\\"end\\":11,\\"type\\":\\"Text\\",\\"content\\":\\" \\"},{\\"tagName\\":\\"Mustache\\",\\"start\\":11,\\"end\\":16,\\"children\\":[],\\"attribute\\":[],\\"content\\":\\"{def}\\",\\"expression\\":{\\"type\\":\\"Identifier\\",\\"start\\":0,\\"end\\":3,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":0},\\"end\\":{\\"line\\":1,\\"column\\":3}},\\"name\\":\\"def\\"}}],\\"attribute\\":[]}],\\"end\\":22,\\"attribute\\":[],\\"content\\":\\"<div>{abc} {def}</div>\\"}"`;

exports[`add function parse simple html 3`] = `"{\\"tagName\\":\\"html\\",\\"start\\":0,\\"children\\":[{\\"start\\":0,\\"end\\":32,\\"tagName\\":\\"div\\",\\"content\\":\\"<div name=\\\\\\"test1 {test2}\\\\\\"></div>\\",\\"children\\":[],\\"attribute\\":[{\\"start\\":5,\\"attrName\\":\\"name\\",\\"attrValue\\":[{\\"start\\":0,\\"end\\":6,\\"type\\":\\"Text\\",\\"content\\":\\"test1 \\"},{\\"tagName\\":\\"Mustache\\",\\"start\\":6,\\"end\\":13,\\"children\\":[],\\"attribute\\":[],\\"content\\":\\"{test2}\\",\\"expression\\":{\\"type\\":\\"Identifier\\",\\"start\\":0,\\"end\\":5,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":0},\\"end\\":{\\"line\\":1,\\"column\\":5}},\\"name\\":\\"test2\\"}}],\\"end\\":25}]}],\\"end\\":32,\\"attribute\\":[],\\"content\\":\\"<div name=\\\\\\"test1 {test2}\\\\\\"></div>\\"}"`;

exports[`add function parse simple html 4`] = `"{\\"tagName\\":\\"html\\",\\"start\\":0,\\"children\\":[{\\"start\\":0,\\"end\\":46,\\"tagName\\":\\"div\\",\\"content\\":\\"<div name=\\\\\\"test1 { a && b ? 'c' : '' }\\\\\\"></div>\\",\\"children\\":[],\\"attribute\\":[{\\"start\\":5,\\"attrName\\":\\"name\\",\\"attrValue\\":[{\\"start\\":0,\\"end\\":6,\\"type\\":\\"Text\\",\\"content\\":\\"test1 \\"},{\\"tagName\\":\\"Mustache\\",\\"start\\":6,\\"end\\":27,\\"children\\":[],\\"attribute\\":[],\\"content\\":\\"{ a && b ? 'c' : '' }\\",\\"expression\\":{\\"type\\":\\"ConditionalExpression\\",\\"start\\":1,\\"end\\":18,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":1},\\"end\\":{\\"line\\":1,\\"column\\":18}},\\"test\\":{\\"type\\":\\"LogicalExpression\\",\\"start\\":1,\\"end\\":7,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":1},\\"end\\":{\\"line\\":1,\\"column\\":7}},\\"left\\":{\\"type\\":\\"Identifier\\",\\"start\\":1,\\"end\\":2,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":1},\\"end\\":{\\"line\\":1,\\"column\\":2}},\\"name\\":\\"a\\"},\\"operator\\":\\"&&\\",\\"right\\":{\\"type\\":\\"Identifier\\",\\"start\\":6,\\"end\\":7,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":6},\\"end\\":{\\"line\\":1,\\"column\\":7}},\\"name\\":\\"b\\"}},\\"consequent\\":{\\"type\\":\\"Literal\\",\\"start\\":10,\\"end\\":13,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":10},\\"end\\":{\\"line\\":1,\\"column\\":13}},\\"value\\":\\"c\\",\\"raw\\":\\"'c'\\"},\\"alternate\\":{\\"type\\":\\"Literal\\",\\"start\\":16,\\"end\\":18,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":16},\\"end\\":{\\"line\\":1,\\"column\\":18}},\\"value\\":\\"\\",\\"raw\\":\\"''\\"}}}],\\"end\\":39}]}],\\"end\\":46,\\"attribute\\":[],\\"content\\":\\"<div name=\\\\\\"test1 { a && b ? 'c' : '' }\\\\\\"></div>\\"}"`;
